src/base1/cockpit.js

```diff
@@ -2145,7 +2160,24 @@ function basic_scope(cockpit, jquery) {
     }
 
     cockpit.logout = function logout(reload) {
-        window.sessionStorage.clear();
+        var i, k;
+        /* Instead of clearing, purge anything that isn't prefixed with an application
+         * and anything prefixed with our application.
```

Comment:
This underscores the need for a helper method here for setting these items. It should be implemented near this code. So that it's one logical concept, when someone comes by trying to understand it.